.buildkite/commands/install-node-dependencies.sh

```diff
@@ -34,13 +34,19 @@ echo "--- :npm: Save cache if necessary"
 # Notice that we don't cache the local node_modules.
 # Those get regenerated by npm ci as per Node reccomendations.
 # What we are caching is the root npm folder, which stores pacakge downloads so they are available if the package.json resolution demands them.
+#
+# npm stores temporary files in ~/.npm that we don't want to extract because they might run into naming conflicts.
+# So, before archiving it, we remove those tmp files.
+#
+# Example: https://buildkite.com/automattic/gutenberg-mobile/builds/8857#018e37eb-7afc-4280-b736-cba76f02f1a3/524
+rm -rf "$HOME/.npm/_cacache/tmp"
```

Comment:
I looked for docs about the role of the `tmp` folder but came out short.

Visited:

- https://docs.npmjs.com/cli/v9/configuring-npm/folders#cache
- https://docs.npmjs.com/cli/v9/commands/npm-cache
- https://www.npmjs.com/package/cacache

I still think it's safe to remove the `tmp` folder.  As the name suggests, it should only contain temporary files.

Reply:
I also looked into my local `pnpm` cache for an equivalent, but couldn't find it.

I assume if something with the possibility of name clashing exists there, it'll reveal itself at some point in CI, like the `~/.npm` one did.

Reply:
> I still think it's safe to remove the tmp folder. As the name suggests, it should only contain temporary files.

I would agree. The [cacache docs on tmp](https://github.com/npm/cacache?tab=readme-ov-file#--tmpwithtmpcache-opts-cb---promise) suggest that the tmp directory may even be designed to delete itself, so it seems safe to programatically do so. But, collision possibilities would reveal themselves in CI, as you said.